### ISSUE TITLE
Proposal: remove broken syntax for tuple types

### ISSUE THREAD

**Issue by avh4** [elm-lang/elm-compiler#1236]
It is currently possible to use the tuple constructor functions in types.  For example, the following are equivalent:

```elm
f : (,,) Int Char String
f : (Int, Char, String)
```

However, the syntax is broken in that it does not enforce the number of arguments.  The following are also treated as equivalent by the compiler:

```elm
f : (,) Int Char String
f : (,,,,,,,,) Int Char String
```

The syntax is also broken in that it allows you to tell the compiler you want a tuple-1 which is not valid in Elm:

```elm
f : (,) String
f = "a"
```

yields the following error:

```
TYPE MISMATCH
The type annotation for `f` does not match its definition.

2| f : (,) String
The type annotation is saying:

    ( String )

But I am inferring that the definition has this type:

    String
```

The syntax is also broken in that it produces an error message exposing compiler internals with a large number of commas:

```elm
f : (,,,,,,,,) Int
```

yields the following error:

```
NAMING ERROR
Cannot find type `_Tuple10`

2| f : (,,,,,,,,,) Int
Maybe you want one of the following?

    _Tuple0
    _Tuple1
    _Tuple2
    _Tuple3
```

In summary, this syntax is not documented, unlikely to be widely known and used, and does not function in a desirable way.  I propose that this syntax should be removed, as there is an existing, widely-known, and correctly-functioning alternative way to represent tuple types e.g., `(Int, Char, String)`.

**Comment by mgold** [comment]
Whoa, I didn't even know that was possible! And the face that it doesn't work implies that it was never meant to exist. So yes, it should be removed. That said, given how (almost?) no one has run into this before, I'm not sure how high a priority it is, and how difficult it would be.

**Comment by avh4** [comment]
This should be pretty easy.  I could make the PR if it is agreed that this should be done.

**Comment by laszlopandy** [comment]
Remove it. No objections.

**Comment by evancz** [comment]
I think I needed it for type annotations when the pretty printer was way worse or something. I don't recall the exact circumstances, but for some reason I needed this to exist. I'm down to remove it, but I don't know if it'll have surprise implications.

**Comment by evancz** [comment]
Follow along in #1375
